### PR TITLE
[FLINK-7134] Remove hadoop1.x code in mapreduce.utils.HadoopUtils

### PR DESCRIPTION
**Pull request by zhangminglei**

### CODE REVIEW

File: flink-connectors/flink-hcatalog/src/main/java/org/apache/flink/hcatalog/HCatInputFormatBase.java [hunk 1]
```diff
@@ -271,7 +273,7 @@ public BaseStatistics getStatistics(BaseStatistics cachedStats) throws IOExcepti
 
 		JobContext jobContext = null;
 		try {
-			jobContext = HadoopUtils.instantiateJobContext(configuration, new JobID());
+			jobContext = new JobContextImpl(configuration, new JobID());
```
Comment (StephanEwen) [inline flink-connectors/flink-hcatalog/src/main/java/org/apache/flink/hcatalog/HCatInputFormatBase.java]: We can remove the try/catch block here...

File: flink-connectors/flink-hcatalog/src/main/java/org/apache/flink/hcatalog/HCatInputFormatBase.java [hunk 2]
```diff
@@ -299,7 +301,7 @@ public InputSplitAssigner getInputSplitAssigner(HadoopInputSplit[] inputSplits)
 	public void open(HadoopInputSplit split) throws IOException {
 		TaskAttemptContext context = null;
 		try {
-			context = HadoopUtils.instantiateTaskAttemptContext(configuration, new TaskAttemptID());
+			context = new TaskAttemptContextImpl(configuration, new TaskAttemptID());
```
Comment (StephanEwen) [inline flink-connectors/flink-hcatalog/src/main/java/org/apache/flink/hcatalog/HCatInputFormatBase.java]: Same as above...

File: flink-connectors/flink-hcatalog/src/main/java/org/apache/flink/hcatalog/HCatInputFormatBase.java (hunk 1 above)
Comment (zhangminglei) [inline flink-connectors/flink-hcatalog/src/main/java/org/apache/flink/hcatalog/HCatInputFormatBase.java]: Sorry. Stephan. I just forgot about it.

**Comment by zhangminglei** [comment]
Hey, Stephan. This time codes works well. Thanks ~ :)

**Pull request closed without merging**